### PR TITLE
Add volumes for Jenkins slaves

### PR DESCRIPTION
**Pull request by icoloma**
This PR adds volume support for slave nodes in Jenkins using Kubernetes. Most volume types available in KubernetesPlugin are available, my particular pain point was using secrets from my Jenkinsfile:

values.yaml
```yaml
Agent:
  Enabled: true
  Image: gcr.io/cloud-solutions-images/jenkins-k8s-slave
  ImageTag: latest
  AlwaysPullImage: true
  volumes:
  - type: Secret
    secretName: jenkins-demo-secrets
    mountPath: /var/run/secrets/jenkins-demo-secrets
```

The secret can be accessed from the `mountPath` in the Jenkins pipeline.

Jenkinsfile (example with gcloud):
```bash
sh "gcloud auth activate-service-account my-service-account@developer.gserviceaccount.com --key-file=/var/run/secrets/jenkins-demo-secrets/my-service-account.json"
```

This PR also adds a `AlwaysPullImage` value, otherwise development with custom slave images is painful.

### CODE REVIEW

File: stable/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -37,13 +37,21 @@ data:
...
               <label></label>
               <nodeSelector></nodeSelector>
-              <volumes/>
+              <volumes>
+{{- range $index, $volume := .Values.Agent.volumes }}
+                <org.csanchez.jenkins.plugins.kubernetes.volumes.{{ $volume.type }}Volume>
+{{- range $key, $value := $volume }}{{- if not (eq $key "type") }}
+                  <{{ $key }}>{{ $value }}</{{ $key }}>
+{{- end }}{{- end }}
+                </org.csanchez.jenkins.plugins.kubernetes.volumes.SecretVolume>
```
Comment (lachie83) [inline stable/jenkins/templates/config.yaml]: Should this be `</org.csanchez.jenkins.plugins.kubernetes.volumes.{{ $volume.type }}Volume>`

**Comment by lachie83** [comment]
@icoloma Can you please sign the CLA?

File: stable/jenkins/templates/config.yaml (hunk 1 above)
Comment (viglesiasce) [reply stable/jenkins/templates/config.yaml]: Nice catch @lachie83

**Comment by lachie83** [comment]
@icoloma CLA still needs signing

**Comment by icoloma** [comment]
I have been trying to sign, but the process is broken for me. I have sent a support request at the CNCF portal, but nobody contacted back. Do you know how I could take this offline?

**Comment by icoloma** [comment]
@lachie83 I have been trying to sign, but the process is broken for me. I have sent a support request at the CNCF portal, but nobody got back to me. Do you know who I could contact for this?

**Comment by lachie83** [comment]
@icoloma Any updates from support?

**Comment by lachie83** [comment]
@viglesiasce any ideas on how to fix this?

**Comment by icoloma** [comment]
No updates from support. I am searching for other contacts inside CNCF.

**Comment by lachie83** [comment]
Bummer okay

**Comment by icoloma** [comment]
@lachie83 CLA has been signed :smile: :+1:

**Comment by icoloma** [comment]
@lachie83 Just merged master to the best of my knowledge

**Comment by lachie83** [comment]
@k8s-bot ok to test

**Comment by lachie83** [comment]
Can you bump the Chart version please?

**Comment by lachie83** [comment]
Also please update the README with the new parameters.

**Comment by icoloma** [comment]
@lachie83 Done.

**Comment by viglesiasce** [comment]
Merged in #1385

**Pull request closed without merging**